### PR TITLE
[api-draft][connector] apache pulsar source

### PR DESCRIPTION
**Pull request by ashulin**
## Purpose of this pull request

apache pulsar source

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by Hisoka-X** [comment]
Please add license to LICENSE file in seatunnel-dist

**Comment by BenJFan** [comment]
Please add license to LICENSE file in seatunnel-dist

**Comment by Hisoka-X** [comment]
Also add pulsar connector into pom.xml in `seatunnel-connector-seatunnel-dist`

**Comment by BenJFan** [comment]
Also add pulsar connector into pom.xml in `seatunnel-connector-seatunnel-dist`

**Comment by ashulin** [comment]
@BenJFan I have added the licenses.

**Review by CalvinKirs — COMMENTED** [review]
Some Notices also need to be added.  https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/release-docs/NOTICE

**Comment by ashulin** [comment]
@CalvinKirs Do I need to add any licenses or notices?

File: tools/dependencies/known-dependencies.txt [hunk 1]
```diff
@@ -50,6 +50,10 @@ aws-java-sdk-core-1.12.37.jar
 aws-java-sdk-ec2-1.12.37.jar
 aws-java-sdk-kms-1.12.37.jar
 aws-java-sdk-s3-1.12.37.jar
+bcpkix-jdk15on-1.68.jar
+bcprov-ext-jdk15on-1.68.jar
+bcprov-jdk15on-1.68.jar
+bouncy-castle-bc-2.8.0-pkg.jar
```
Comment (CalvinKirs) [inline tools/dependencies/known-dependencies.txt:56]: Are these licenses also lost? I don't found in https://github.com/apache/incubator-seatunnel/pull/1984/files#diff-01f40f6b66e748f1845983d2ebae82ebcdc07c318bb0ee5ddab1fad666a2f859

Comment (ashulin) [reply tools/dependencies/known-dependencies.txt:56]: `bouncy-castle-bc-2.8.0-pkg.jar` is module of the `apache pulsar` project.
**LICENSE-bouncycastle.txt** is the license file for other dependencies.

Comment (CalvinKirs) [reply tools/dependencies/known-dependencies.txt:56]: bcpkix-jdk15on-1.68 and bcprov-ext-jdk15on-1.68 etc...  is not reflected in https://github.com/apache/incubator-seatunnel/pull/1984/files#diff-01f40f6b66e748f1845983d2ebae82ebcdc07c318bb0ee5ddab1fad666a2f859 , could check again?

Comment (ashulin) [reply tools/dependencies/known-dependencies.txt:56]: I added bouncycastle's dependencies to the MIT license.

**Comment by wachoo** [comment]
I run this command with the jar "connector-pulsar-2.3.3.jar"

```shell
./bin/seatunnel.sh --config ./config/v2.streaming.pulsar2console.template -e local
```
image


the connection  to admin and client is ok, but it can not consume the messages .

the connection configuration below like this:

```shell
env {
    "job.mode"=STREAMING
    "job.name"="seatunnel_schema2console_test"
}
source {
    Pulsar {
        "subscription.name"="seatunnel_schema2console_test.sub"
        "cursor.startup.mode"=LATEST
        "cursor.reset.mode"=null
        "cursor.stop.mode"=NEVER
        "topic-discovery.interval"=-1
        "poll.timeout"=100
        "poll.interval"=50
        "poll.batch.size"=500
        parallelism=1
        "result_table_name"=Table11621381770080
        topic="persistent://public/default/seatunnel.schema.test"
        schema {
            fields {
                id=string
            }
        }
        operationTimeoutMs="30000"
        useTls="false"
        connectionMaxIdleSeconds="30000"
        useTcpNoDelay="true"
        "client.service-url"="pulsar://xxx.xxx.xxx.xxx:6650"
        "admin.service-url"="http://xxx.xxx.xxx.xxx:8080"
        topic-pattern="false"
        lookupTimeoutMs="30000"
        tlsHostnameVerificationEnable="true"
    }
}
transform {
}
sink {
    Console {
    }
}
```

is  there some demo for pulsar?

**Pull request merged**